clients/drcachesim/analyzer.cpp

```diff
@@ -295,6 +296,21 @@ analyzer_tmpl_t<RecordType, ReaderType>::init_scheduler_common(
     std::vector<typename sched_type_t::input_workload_t> &workloads,
     typename sched_type_t::scheduler_options_t options)
 {
+    // Add noise generator to workload_inputs.
+    if (noise_generator_factory_.is_enabled()) {
+        // TODO i#7216: here can be a good place to analyze the workloads in order to
+        // tweak  noise_generator_info_t parameters. We'd also need a method to replace
+        // noise_generator_info_t in noise_generator_factory_t.
+        noise_generator_factory_.add_noise_generator_to_workloads(workloads);
+        // Check for errors (e.g., a record_file_reader_t noise generator).
+        std::string noise_generator_factory_error_string =
+            noise_generator_factory_.get_error_string();
+        if (!noise_generator_factory_error_string.empty()) {
+            error_string_ = noise_generator_factory_error_string;
```

Comment:
Just assign this instead of using the local var?  If we don't want to clobber a pre-existing error: this is already clobbering it; could append.

Reply:
Done (with append).